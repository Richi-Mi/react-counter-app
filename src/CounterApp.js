// rafcp
import React, { useState } from 'react';
import PropTypes from 'prop-types';

const CounterApp = ({ value }) => {
    let [ counter, setCounter ] = useState( value ) // Retorna un arreglo con 2 valores
    
    // addNumber
    const handleAdd = () => {
        setCounter( counter + 1 )
        // setCounter( variable => variable + 1 )
    }
    const handleSubstract = () => {
        setCounter( counter - 1 )
    }
    const handleReset = () => {
        setCounter( counter = value )
    }

    return (
        < >
            <h1> CounterApp </h1>
            <h2> Number:  { counter } </h2>

            <button onClick={ handleAdd }> Add Number </button>
            <button onClick={ handleReset }> Reset </button>
            <button onClick={ handleSubstract }> Substract Number </button>
        </ >
    );
}
// Poner las funciones bien, sin los parentesis
CounterApp.propTypes = {
    value: PropTypes.number
}
export default CounterApp