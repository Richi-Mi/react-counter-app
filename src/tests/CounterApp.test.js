import { shallow } from "enzyme"
import CounterApp from "../CounterApp"

describe(' - Pruebas sobre el componente CounterApp', () => {
    // let wrapper;
    let wrapper = shallow( <CounterApp /> );
    beforeEach( () => {
        // Esta función va a ejecutar lo que se recibe en el callback, cada que haga una Prueba. en el test()
        wrapper = shallow( <CounterApp /> );
    })
    test('Debe Mostrar el componente Correctamente, con sus valores por defecto', () => {
        // const valueDef = 10;
        // const [ txt, counter ] = wrapper.find('h2').text().split(':')
        
        expect( wrapper ).toMatchSnapshot();
        // expect( parseInt(counter) ).toBe( valueDef )
    })
    test('Debe Mostrar el valor enviado de 100', () => {
        const value = 100;
        const wrapper = shallow( <CounterApp value={ value }/> );
        const [ txt, counter ] = wrapper.find('h2').text().split(':')

        expect( parseInt(counter) ).toBe( value )
    })
    
    test('Debe incrementar con el boton de +1', () => {
        const btn_add = wrapper.find('button').at(0) // at(0) .- Nos sirve para hacer referencia al boton por su posición en el componente
        // console.log( btn_add.html() ) .- Imprime el wrapper en su version de html y devuelve el codigo html
        btn_add.simulate('click')
        const counterText = wrapper.find('h2').text().trim()

        expect( counterText ).toBe("Number: 11")
    })
    test('Debe incrementar con el boton de -1', () => {
        // Para hacer esta prueba requerimos reiniciar el componente de Pruebas wrapper
        const btn_substract = wrapper.find('button').at(2)  
        btn_substract.simulate('click')
        const counterText = wrapper.find('h2').text().trim()

        expect( counterText ).toBe("Number: 9")
    })
    test('Debe colocar el valor por defecto con el btn reset', () => {
        const wrapper = shallow( <CounterApp value={ 105 }/> );

        const btn_add = wrapper.find('button').at(0); 
        btn_add.simulate('click');
        btn_add.simulate('click');

        const btn_reset = wrapper.find('button').at(1);
        btn_reset.simulate('click');

        const counterText = wrapper.find('h2').text().trim();

        expect( counterText ).toBe('Number: 105')
    })
})
