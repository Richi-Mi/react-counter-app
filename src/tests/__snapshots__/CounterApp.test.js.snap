// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` - Pruebas sobre el componente CounterApp Debe Mostrar el componente Correctamente, con sus valores por defecto 1`] = `
<Fragment>
  <h1>
     CounterApp 
  </h1>
  <h2>
     Number: 
    10
  </h2>
  <button
    onClick={[Function]}
  >
     Add Number 
  </button>
  <button
    onClick={[Function]}
  >
     Reset 
  </button>
  <button
    onClick={[Function]}
  >
     Substract Number 
  </button>
</Fragment>
`;
